worlds/apquest/client/custom_views.py

```diff
@@ -0,0 +1,234 @@
+        self._keyboard.bind(on_key_down=self._on_keyboard_down)
+
+    def _keyboard_closed(self) -> None:
+        if self._keyboard is None:
+            return
+        self._keyboard.unbind(on_key_down=self._on_keyboard_down)
+        self._keyboard = None
+
+    def _on_keyboard_down(self, _, keycode, _1, _2) -> bool:
+        if keycode[1] == "up":
+            self.input_function(Input.UP)
+        elif keycode[1] == "down":
+            self.input_function(Input.DOWN)
+        elif keycode[1] == "left":
+            self.input_function(Input.LEFT)
+        elif keycode[1] == "right":
+            self.input_function(Input.RIGHT)
+        elif keycode[1] == "spacebar":
+            self.input_function(Input.ACTION)
+        elif keycode[1] == "c":
+            self.input_function(Input.CONFETTI)
+        return True
+
+
+class APQuestGrid(GridLayout):
+    def check_resize(self, _: int, _1: int) -> None:
+        parent_width, parent_height = self.parent.size
+
+        self_width_according_to_parent_height = parent_height * 12 / 11
+        self_height_according_to_parent_width = parent_height * 11 / 12
+
+        if self_width_according_to_parent_height > parent_width:
+            self.size = parent_width, self_height_according_to_parent_width
+        else:
+            self.size = self_width_according_to_parent_height, parent_height
+
+
+CONFETTI_COLORS = [
+    (220 / 255, 0, 212 / 255),  # PINK
+    (0, 0, 252 / 255),  # BLUE
+    (252 / 255, 220 / 255, 0),  # YELLOW
+    (0, 184 / 255, 0),  # GREEN
+    (252 / 255, 56 / 255, 0),  # ORANGE
+]
+
+
+@dataclass
+class Confetti:
+    x_pos: float
+    y_pos: float
+    x_speed: float
+    y_speed: float
+    color: tuple[float, float, float]
+    life: float = 3
+
+    triangle1: Triangle | None = None
+    triangle2: Triangle | None = None
+    color_instruction: Color | None = None
+
+    def update_speed(self, dt: float):
+        if self.x_speed > 0:
+            self.x_speed -= 2.7 * dt
+            if self.x_speed < 0:
+                self.x_speed = 0
+        else:
+            self.x_speed += 2.7 * dt
+            if self.x_speed > 0:
+                self.x_speed = 0
+
+        if self.y_speed > -0.03:
+            self.y_speed -= 2.7 * dt
+            if self.y_speed < -0.03:
+                self.y_speed = -0.03
+        else:
+            self.y_speed += 2.7 * dt
+            if self.y_speed > -0.03:
+                self.y_speed = -0.03
+
+    def move(self, dt: float):
+        self.update_speed(dt)
+
+        if self.y_pos > 1:
+            self.y_pos = 1
+            self.y_speed = 0
+        if self.x_pos < 0.01:
+            self.x_pos = 0.01
+            self.x_speed = 0
+        if self.x_pos > 0.99:
+            self.x_pos = 0.99
+            self.x_speed = 0
+
+        self.x_pos += self.x_speed * dt
+        self.y_pos += self.y_speed * dt
+
+    def render(self, offset_x, offset_y, max_x, max_y) -> tuple[list[float], list[float]]:
```

Comment:
```suggestion
    def render(self, offset_x, offset_y, max_x, max_y) -> None:
```
Doesn't return anything.